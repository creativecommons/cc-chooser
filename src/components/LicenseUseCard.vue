<template>
    <div class="license-use-card">
        <h4 class="vocab b-header">
            {{ $t('license-use.heading') }}
        </h4>
        <p class="license-use-instructions">
            {{ $t('license-use.common-instructions') }}
        </p>
        <b-tabs>
            <b-tab-item :label="$t('license-use.web-tab-heading')">
                {{ $t('license-use.web-instructions') }}
                <LicenseCopy :is-web="true" />
            </b-tab-item>
            <b-tab-item :label="$t('license-use.print-media-tab-heading')">
                {{ $t('license-use.print-media-instructions') }}
                <LicenseCopy :is-web="false" />
            </b-tab-item>
        </b-tabs>
    </div>
</template>
<script>
import LicenseCopy from './LicenseCopy'
export default {
    name: 'LicenseUseCard',
    components: {
        LicenseCopy
    }
}
</script>
<style lang="scss">
.license-use-card {
    margin-bottom: 24px;
}
.license-use-card .h4b {
    color: black;
}
.tabs span {
    font-weight: bold;
}
.license-use-card .tabs .is-active a {
    margin-bottom: -2px;
    border-bottom: 3px solid #04a434;
    span {
        font-weight: 700;
        opacity:1;
        color: black;
    }
}
</style>
