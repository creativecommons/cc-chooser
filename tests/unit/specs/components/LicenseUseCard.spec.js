import { shallowMount, createLocalVue, config } from '@vue/test-utils'
import LicenseUseCard from '@/components/LicenseUseCard.vue'
import VueI18n from 'vue-i18n'
import Buefy from 'buefy'

describe('LicenseUseCard.vue', () => {
    let wrapper

    // Always creates a shallow instance of component
    beforeEach(() => {
        const localVue = createLocalVue()
        localVue.use(VueI18n)
        localVue.use(Buefy)
        const messages = require('@/locales/en.json')
        const i18n = new VueI18n({
            locale: 'en',
            fallbackLocale: 'en',
            messages: messages,
            silentTranslationWarn: true
        })

        config.mocks.i18n = i18n

        config.mocks.$t = (key) => {
            return i18n.messages[key]
        }
        wrapper = shallowMount(LicenseUseCard, {
            localVue,
            i18n
        })
    })

    it('Check if the LicenseUseCard.vue component renders without any errors', () => {
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    // Test for DOM elements which must be present
    it('Check if the main div tag with the class license-use-card is present', () => {
        expect(wrapper.contains('.license-use-card')).toBe(true)
    })
    it('Check if the main header is present in the DOM', () => {
        expect(wrapper.contains('.license-use-card h4')).toBe(true)
    })
    it('Check if the common instructions are present in the DOM', () => {
        expect(wrapper.contains('.license-use-instructions')).toBe(true)
    })
    it('Check if the tab and the tab items are present in the DOM', () => {
        expect(wrapper.contains('b-tabs-stub')).toBe(true)
        expect(wrapper.contains('b-tab-item-stub')).toBe(true)
        expect(wrapper.findAll('b-tab-item-stub').length).toBe(2)
    })

    // Snapshot tests
    it('LicenseUseCard.vue component has the expected UI', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
