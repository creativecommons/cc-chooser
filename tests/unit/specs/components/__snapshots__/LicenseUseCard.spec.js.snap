// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseUseCard.vue LicenseUseCard.vue component has the expected UI 1`] = `
<div class="license-use-card">
  <h4 class="vocab h4b hb">

  </h4>
  <p class="license-use-instructions">

  </p>
  <b-tabs-stub animated="true">
    <b-tab-item-stub visible="true">

      <licensecopy-stub isweb="true"></licensecopy-stub>
    </b-tab-item-stub>
    <b-tab-item-stub visible="true">

      <licensecopy-stub></licensecopy-stub>
    </b-tab-item-stub>
  </b-tabs-stub>
</div>
`;
